pkg/controller/config/match/match.go

```diff
@@ -0,0 +1,65 @@
+package match
+
+import (
+	configv1alpha1 "github.com/open-policy-agent/gatekeeper/apis/config/v1alpha1"
+)
+
+type Operation string
+
+const (
+	Audit   = Operation("audit")
+	Sync    = Operation("sync")
+	Webhook = Operation("webhook")
+	Star    = Operation("*")
+)
+
+type Set struct {
+	ExcludedNamespaces map[Operation][]string
+}
+
+var allOperations = []Operation{
+	Audit,
+	Webhook,
+	Sync,
+}
+
+var configMapSet = &Set{
+	ExcludedNamespaces: make(map[Operation][]string),
+}
```

Comment:
instead of a list of strings, this could be: `map[Operation]map[string]bool`

This would make checking namespace membership in a given operation O(1) and avoid the need for accounting work like making sure a namespace isn't added twice.

If we do this, we no longer need to pre-prepare a list before every execution.

Reply:
updated to `map[Operation]map[string]bool`